pkgs/by-name/bl/black-hole-solver/package.nix

```
@@ -21,24 +20,23 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [
     cmake
-    perl
+    (buildPackages.perl.withPackages (ps: [ ps.PathTiny ]))
```

```suggestion
    perl.withPackages (ps: [ ps.PathTiny ])
```

I guess this should be sufficient because packages in `nativeBuildInputs` are chosen `pkgsBuildHost` in slicing?

Or you want this?

```
  depsBuildBuild = [
    perl.withPackages (ps: [ ps.PathTiny ])
  ];
```

---

I think using `withPackages` breaks splicing, but I will check again.

---

just tested the suggested change: `perl.withPackages ...` produces a perl for the host platform, even if the result is in `depsBuildBuild`.